### PR TITLE
Scan after 2FA

### PR DESCRIPTION
**Pull request by RafaRuiz**
Scan after 2FA

Warning. If originally chosen QR scan, and the 2FA dialog is shown, it'll go to the default scan type (`NFC`).

The reason why I'm bypassing this is because Flutter and React Native can't choose the way it'll scan, it'll just "scan" with the default method. If NFC isn't enabled, it'll automatically go to QR scan.

Android users should `Doordeck.withEventsCallback`  as this change is optional, and the sample app uses it in the `unlockCallback`

### CODE REVIEW

File: ui/src/main/java/com/doordeck/sdk/common/manager/Doordeck.kt [hunk 1]
```diff
@@ -316,8 +316,8 @@ object Doordeck {
     /**
      * Shows the verification screen
      */
-    fun showVerificationScreen(context: Context) {
-        VerifyDeviceActivity.start(context)
+    fun showVerificationScreen(context: Context, shouldUnlockAfterSuccessVerifying: Boolean) {
```
Comment (mpbarnwell) [inline ui/src/main/java/com/doordeck/sdk/common/manager/Doordeck.kt:319]: Do we need to add a default value to `shouldUnlockAfterSuccessVerifying` as well? I don't know if we're expecting developers to call this directly or not, hence how much it affects upgrading.

File: ui/src/main/java/com/doordeck/sdk/ui/verify/VerifyDeviceActivity.kt [hunk 2]
```diff
@@ -206,7 +207,12 @@ internal class VerifyDeviceActivity : BaseActivity(), VerifyDeviceView {
 
     // the code is valid, close the with and come back on the previous screen : The UnlockActivity
     override fun succeed() {
-        finish()
+        if (shouldUnlockAfterSuccessVerifying) {
+            finish()
+            Doordeck.showUnlock(this)
+        } else {
+            finish()
+        }
```
Comment (mpbarnwell) [inline ui/src/main/java/com/doordeck/sdk/ui/verify/VerifyDeviceActivity.kt:215]:
```suggestion
        finish()
        if (shouldUnlockAfterSuccessVerifying) {
            Doordeck.showUnlock(this)
        }
```

**Review by mpbarnwell — APPROVED** [review]

File: ui/src/main/java/com/doordeck/sdk/ui/verify/VerifyDeviceActivity.kt (hunk 2 above)
Comment (RafaRuiz) [reply ui/src/main/java/com/doordeck/sdk/ui/verify/VerifyDeviceActivity.kt]: Pushed, thanks.
That's me testing order stuff :D and committing what worked

File: ui/src/main/java/com/doordeck/sdk/common/manager/Doordeck.kt (hunk 1 above)
Comment (RafaRuiz) [reply ui/src/main/java/com/doordeck/sdk/common/manager/Doordeck.kt:319]: It has it indeed, but at `VerifyDeviceActivity.start`

This method is also public so I'm pushing it now

**Pull request merged**